iina/FilterPresets.swift

```diff
@@ -115,6 +115,30 @@ class FilterParameter {
   }
 }
 
+/// An ordered collection of name to `FilterParameter` pairs.
+///
+/// This structure preserves the order of filter parameters specified in the `FilterPreset` definitions by using [KeyValuePairs](https://developer.apple.com/documentation/swift/keyvaluepairs)
+/// instead of a Swift dictionary to hold the name to `FilterParameter` mapping. The order of the parameters dictates the order
+/// of the UI controls displayed to the user. A dictionary **must not** be used as that would cause the UI controls representing the
+/// parameters to appear in random order.
+struct FilterParameters {
```

Comment:
Why should we create a new struct for this? I think we can directly use `KeyValuePairs<String, FilterParameter>` for `FilterPreset.params`. For the `.get()` method, we can implement in the extension of `KeyValuePairs`

Reply:
Not possible to implement the `get` method as an extension to `KeyValuePairs` because the `Key` generic parameter is not defined as being equatable. No matter. The approach I used is totally and completely wrong. I missed that there is already a `paramOrder`. But is it optional and missing from several presets. I'm closing this PR and opening a new one that uses the existing system for ordering parameters.